buildpacks/cnb.yaml

```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: build.knative.dev/v1alpha1
+kind: BuildTemplate
+metadata:
+  name: cnb
+spec:
+  parameters:
+  - name: IMAGE
+    description: The image you wish to create. For example, "repo/example", or "example.com/repo/image".
+  - name: RUN_IMAGE
+    description: The run image buildpacks will use as the base for IMAGE.
+    default: packs/run
+  - name: USE_CRED_HELPERS
+    description: Use Docker credential helpers for Google's GCR, Amazon's ECR, or Microsoft's ACR.
+    default: 'true'
+
+  steps:
+  - name: prepare
+    image: packs/base
```

Comment:
I don't love all of these being `:latest` and `imagePullPolicy: Always`. That gives the `packs` owners the ability to break this BuildTemplate inadvertently by pushing new images. WDYT about versioning these by digest? There's also a small perf win if you use digests + `imagePullPolicy: IfNotPresent` in not having to ping dockerhub to see if the image has changed.

As of now:

```
packs/base@sha256:0d6949aaa62649223ca13c8a02a03584068e1e5bdd8ed6edd992e107603b083c
packs/build@sha256:99bcaa9904d7eef78df9c04ddf22dc9d78b0d06d548c68c03a71d04704df3048
packs/util@sha256:f97205c746358d03b4aa84480335f04cd69b25a0348f85a6a5a1c7e9aa7f2a86
```

OTOH, that means you can't fix things in the wild, but I tend to favor reproducibility /shrug

Reply:
FWIW, Digests will work better with the caching model I'm adding to BuildTemplates and provide a much better story for release signoff on a particular shape of buildtemplate.

Reply:
We could use digests for `packs/util` because it contains utilities that won't change often, but it is more problematic for `packs/build`. 

Part of the value prop of buildpacks is that they keep your application and system dependencies up to date to avoid CVEs. A buildpack version comes with a tested set of dependencies (for an example look at http://buildpacks.cloudfoundry.org/#/buildpacks, these are CF buildpacks but the Cloud Native Buildpacks will have the same dependency model). New buildpack versions are released regularly with updated dependencies. When new buildpack versions are released we need to bump the buildpacks in the `packs/build` image. 

This means we will be releasing `packs/build` frequently. If we pinned the image to a digest we would need to make an awful lot of PRs, and users would need to constantly reapply the template to get desired security benefits.

Reply:
This is also intrinsic to the CNB model -- we build directly on the build image so that we don't have to nest containers. That means any dependencies that are statically linked into the app binaries won't get CVE updates until you re-build against the latest build image.

Eventually, we may consider using tags to provide buildpack/OS-package updates without changing the CNB code itself. But for now, we'd like each build to use the latest build image.

Reply:
> Part of the value prop of buildpacks is that they keep your application and system dependencies up to date to avoid CVEs.

Ah, this makes lots of sense, ignore me! 👍